EngineLayer/ModernSearch/ModernSearchEngine.cs

```diff
@@ -157,58 +157,49 @@ protected override MetaMorpheusEngineResults RunSpecific()
             return new MetaMorpheusEngineResults(this);
         }
 
-        #endregion Protected Methods
-
-        #region Private Methods
-
-        private void CalculatePeptideScores(IMsDataScan<IMzSpectrum<IMzPeak>> spectrum, double[] peptideScores, double thePrecursorMass)
+        protected void CalculatePeptideScores(IMsDataScan<IMzSpectrum<IMzPeak>> spectrum, double[] peptideScores, double thePrecursorMass)
         {
-            if (!addCompIons)
+            List<int> listSeenipos = new List<int>();
+            for (int i = 0; i < spectrum.MassSpectrum.Size; i++)
             {
-                for (int i = 0; i < spectrum.MassSpectrum.Size; i++)
-                {
-                    var theAdd = 1 + spectrum.MassSpectrum[i].Intensity / spectrum.TotalIonCurrent;
-                    var experimentalPeakInDaltons = spectrum.MassSpectrum[i].Mz - Constants.protonMass;
-                    GeneratePeptideScores(theAdd, experimentalPeakInDaltons, peptideScores);
-                }
+                var theAdd = 1 + spectrum.MassSpectrum[i].Intensity / spectrum.TotalIonCurrent;
+                var experimentalPeakInDaltons = spectrum.MassSpectrum[i].Mz - Constants.protonMass;
+                GeneratePeptideScores(theAdd, experimentalPeakInDaltons, peptideScores, listSeenipos);
             }
-            else
+            listSeenipos.Clear();
```

Comment:
Allows double peak counting if one is seen in original spectra and one in complementary ions

Reply:
please move the clear to the if statement below